### PR TITLE
Introduce OpenSUSE containerdisks

### PR DESCRIPTION
**Pull request by lyarwood**
/cc @0xFelix 

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Containerdisks for OpenSUSE Leap and Tumbleweed have been added to the project.
```

### CODE REVIEW

**Comment by lyarwood** [comment]
/test pull-containerdisks-build

https://quay.io/repository/containerdisks/opensuse-tumbleweed
https://quay.io/repository/containerdisks/opensuse-leap

Both exist now so this should hopefully pass.

File: artifacts/opensuse/tumbleweed/tumbleweed.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+		return nil, fmt.Errorf("error downloading the tumbleweed SHA256SUMS file: %v", err)
+	}
+	checksums, err := hashsum.Parse(bytes.NewReader(raw), hashsum.ChecksumFormatGNU)
+	if err != nil {
+		return nil, fmt.Errorf("error reading the SHA256SUMS file: %v", err)
+	}
+
+	for file, checksum := range checksums {
+		// openSUSE-Tumbleweed-Minimal-VM.x86_64-1.0.0-kvm-and-xen-Snapshot20240629.qcow2
+		if regexp.MustCompile(fmt.Sprintf("%s\\.%s\\-1\\.0\\.0\\-%s", t.Variant, t.Arch, t.SubVariant)).MatchString(file) {
```
Comment (0xFelix) [inline artifacts/opensuse/tumbleweed/tumbleweed.go:46]: Compile this regex just once outside of the for loop?

File: artifacts/opensuse/leap/leap.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+	return []api.ArtifactTest{
+		tests.SSH,
+	}
+}
+
+func New(arch, version string) *leap {
+	return &leap{
+		Arch:    arch,
+		Version: version,
+		getter:  &http.HTTPGetter{},
```
Comment (0xFelix) [inline artifacts/opensuse/leap/leap.go:79]: No `envVariables` for LEAP?

Comment (lyarwood) [reply artifacts/opensuse/leap/leap.go:79]: Not yet, I'll add in a follow up commit once we have preferences in common-instancetypes.

**Comment by lyarwood** [comment]
[Updated](https://github.com/kubevirt/containerdisks/compare/5982c636f9d585351d44c91b8e591a38c4c20539..2d7077ecf7f44bcd67074c35f5e5597041639978) I've dropped EnvVariables entirely from tumbleweed for now, will add for both in a follow up once we had preferences.

**Review by 0xFelix — APPROVED** [review]
/approve
/lgtm

**Pull request merged**